app/controllers/wpcc/your_results_controller.rb

```diff
@@ -21,6 +21,14 @@ def salary_frequency
     end
     helper_method :salary_frequency
 
+    def your_results_presenter
+      @your_results_presenter ||= Wpcc::YourResultsPresenter.new(
+        OpenStruct.new(eligible_salary: session[:eligible_salary]),
+        view_context: view_context
+      )
+    end
+    helper_method :your_results_presenter
```

Comment:
We now have two ways of consuming presenter methods. Which one should we use in the long term?

Reply:
If I understand this correctly, I think @tomas-stefano is using the OpenStruct class to enable him to pass the eligible_salary stored in the session, **which is a string**, to the presenter. All presenters need to be instantiated with an **object** and a view_context. Using OpenStruct in this manner, creates an object which has the attribute eligible_salary which can be called in the presenter, or in this case in the EarningsDescription module.